### DISCUSSION TITLE
Error iterating nodes

### DISCUSSION THREAD

**Discussion by lunicirus** [pyswmm/pyswmm#408] [Q&A] [answered]
Hi,

I am new in pyswmm and I'm trying to iterate all nodes. However, after a couple of nodes I get the error "error in method 'project_get_index', argument 2 of type 'char *". 
I can execute the network without problems. 

Thank you in advance for any help provided.

**Comment by bemcdonnell** [comment, marked as answer]
@lunicirus can you please share your code here?

**Reply by lunicirus** [reply]
Yes of course. ITs very simple:

`sim = Simulation('01-Network.inp')
sim.execute()

node_object = Nodes(sim)

for node in Nodes(sim):
    print(node.nodeid)`

**Reply by bemcdonnell** [reply]
Remove sim.execute(). Use the “with” context manager to open the model instead.

**Reply by lunicirus** [reply]
Thank you, I did but I still get the same error. 

<img width="607" alt="Screen Shot 2023-03-21 at 3 45 23 PM" src="https://user-images.githubusercontent.com/2193587/226736583-fcda6bb8-cfcb-49a7-bbcd-21bd2a18b0e3.png">

**Reply by fileops** [reply]
What are the names of all the nodes in your model? If there's a lot, what might be the next in the series?

**Reply by bemcdonnell** [reply]
@lunicirus if you take out the node iterator and run it, will it run?  Are you by chance trying to run another simulation at the same time?  I’m wondering if there is some weird character?

**Reply by bemcdonnell** [reply]
@lunicirus also one more important question - which version of pyswmm are you using?

**Reply by lunicirus** [reply]
@bemcdonnell do you mean run the code removing the "for node..." part? it runs fine without that. The version is 1.2 I think.

**Reply by lunicirus** [reply]
This is the first part of the list of junctions. Maybe is due to the comments (starting with ";") in between the lines?
<img width="514" alt="Screen Shot 2023-03-21 at 4 35 43 PM" src="https://user-images.githubusercontent.com/2193587/226747994-0c2895bb-629a-456e-ab9e-6063b9b93378.png">

**Reply by fileops** [reply]
It is likely the à character.

**Reply by bemcdonnell** [reply]
Voila. My guess is it’s the character. While SWMM can run it, pyswmm cannot interpret (what I believe is) the Unicode values at this time. However, I would be grateful if someone would want to introduce that capability.

**Reply by lunicirus** [reply]
Thank you @fileops and @bemcdonnell so much for your help!!

**Reply by karosc** [reply]
This is weird, because I am totally able to read unicode node ids with that character in it using pyswmm 1.2.0 on windows 10 and ubuntu linux. Maybe the text file is not encoded with utf-8?

![image](https://user-images.githubusercontent.com/36245370/227066372-2db5ef9f-0bdb-4000-af41-3530dce119f7.png)

![image](https://user-images.githubusercontent.com/36245370/227066402-744726bb-12be-4da3-9ece-286209ec5572.png)

**Reply by fileops** [reply]
sounds like "a grave" problem. A with the grave symbol is represented differently across the different formats. I'm not an expert but from what I can tell it messes with PySWMM.

**Reply by karosc** [reply]
If someone can post a file that reproduces this error in pyswmm 1.2.0, I can try to debug it. I can't seem to reproduce it just simply by adding an A with accent grave.
